### PR TITLE
Add Kueue tests 

### PR DESCRIPTION
**Pull request by ChughShilpa**
[RHOAIENG-1572](https://issues.redhat.com/browse/RHOAIENG-1572)

### CODE REVIEW

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+Teardown Kueue E2E Test Suite
+    [Documentation]    Teardown Kueue E2E Test Suite
+    Log To Console    Uninstall Kueue ...
+    ${return_code}    ${output}    Run And Return Rc And Output    kubectl delete -k "github.com/kubernetes-sigs/kueue/config/default?ref=main"
+    Log To Console    ${output}
+    Should Be Equal As Integers   ${return_code}   0  msg=Error detected while uninstalling kueue
+
+    # Remove label instance-type=on-demand from worker node
+    ${return_code} =    Run And Return Rc    oc label ${WORKER_NODE} instance-type-
```
Comment (sutaakar) [inline ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot:63]: Does this code really remove label from node?

Comment (ChughShilpa) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot:63]: Yes, i have tested it

Comment (sutaakar) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot:63]: Thanks

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    [Arguments]    ${test_name}
+    Log To Console    Running Kueue E2E test: ${test_name}
+    ${return_code}    ${output}    Run And Return Rc And Output    ginkgo --focus-file=${test_name} ${KUEUE_DIR}/test/e2e/singlecluster
+    Log To Console    ${output}
+    Should Be Equal As Integers   ${return_code}   0  msg=${test_name} failed
+
+Enable Visibility Feature Gate
+    [Documentation]    Enable Visibility Feature Gate
+    ${return_code} =    Run And Return Rc    oc patch deployment kueue-controller-manager -n kueue-system --type 'json' -p '[{"op" : "add" ,"path" : "/spec/template/spec/containers/0/args/-" ,"value" : "--feature-gates=VisibilityOnDemand=true"}]'
+    Should Be Equal As Integers  ${return_code}   0   msg=Visiblity feature gate is not enabled
```
Comment (sutaakar) [inline ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot:77]: When I ran the file locally then some tests failed on HTTP 503. It looks like the problem was with Kueue still being restarted when tests were executed.
This helped me locally:
```
    ${return_code} =    Run And Return Rc    oc rollout status deployment/kueue-controller-manager -n kueue-system --timeout=3m
    Should Be Equal As Integers  ${return_code}   0   msg=Kueue rollout failed
```

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    ${result} =    Run Process    oc get nodes -o name | tail -n1
+    ...    shell=true    stderr=STDOUT
+    Set Suite Variable    ${WORKER_NODE}    ${result.stdout}
+    ${return_code} =    Run And Return Rc    oc label ${WORKER_NODE} instance-type=on-demand
+    Should Be Equal As Integers  ${return_code}   0   msg=Fail to label worker node with instance-type=on-demand
+
+Teardown Kueue E2E Test Suite
+    [Documentation]    Teardown Kueue E2E Test Suite
+    Log To Console    Uninstall Kueue ...
+    ${return_code}    ${output}    Run And Return Rc And Output    kubectl delete -k "github.com/kubernetes-sigs/kueue/config/default?ref=main"
```
Comment (sutaakar) [inline ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot:58]:
```suggestion
    ${return_code}    ${output}    Run And Return Rc And Output    kubectl delete -k "github.com/kubernetes-sigs/kueue/config/default?ref=main" --ignore-not-found=true
```

Comment (sutaakar) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot:58]: This should prevent issue with deleting already removed resources.

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        FAIL    Unable to clone kueue repo ${KUEUE_REPO_URL}:${KUEUE_REPO_BRANCH}:${KUEUE_DIR}
+    END
+
+    Log To Console    Install the latest development version of kueue ...
+    ${return_code}    ${output}    Run And Return Rc And Output    kubectl apply --server-side -k "github.com/kubernetes-sigs/kueue/config/default?ref=main"
+    Log To Console    ${output}
+    Should Be Equal As Integers   ${return_code}   0  msg=Error detected while installing kueue
+
+    # Add label instance-type=on-demand on worker node
+    ${result} =    Run Process    oc get nodes -o name | tail -n1
```
Comment (sutaakar) [inline ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot:49]:
```suggestion
    ${result} =    Run Process    oc get nodes -o name --selector=node-role.kubernetes.io/worker | tail -n1
```

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot (hunk 3 above)
Comment (ChughShilpa) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot]: Updated

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot (hunk 4 above)
Comment (ChughShilpa) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot]: Updated

**Review by sutaakar — DISMISSED** [review]

**Comment by ChughShilpa** [comment]
Verified in Jenkins build - rhods-pipeline-dw/213/

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+Suite Setup       Prepare Kueue E2E Test Suite
+Suite Teardown    Teardown Kueue E2E Test Suite
+Library           OperatingSystem
+Library           Process
+Library           OpenShiftLibrary
+
+
+*** Variables ***
+${KUEUE_DIR}            kueue
+${KUEUE_REPO_URL}       %{KUEUE_REPO_URL=https://github.com/kubernetes-sigs/kueue.git}
```
[the remark this answers is not in the document]
Comment (ChughShilpa) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot]: Yes, done and verified in  rhods-pipeline-dw/214/

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+${KUEUE_REPO_BRANCH}    %{KUEUE_REPO_BRANCH=main}
+${JOB_GO_BIN}           %{WORKSPACE=.}/go-bin
+${KUBECONFIG}           %{WORKSPACE=.}/kconfig
+${WORKER_NODE}          ${EMPTY}
+
+
+*** Test Cases ***
+Run E2E test
+    [Documentation]    Run ginkgo E2E single cluster test
+    [Tags]  Kueue
```
Comment (jiripetrlik) [inline ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot:22]: Do you plan to add also tag for tier (like Sanity, Tier2, etc.)?

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    ...     DistributedWorkloads
+    Run Kueue E2E Test    e2e_test.go
+
+Run Visibility test
+    [Documentation]    Run ginkgo visibilty single cluster test
+    [Tags]  Kueue
```
Comment (jiripetrlik) [inline ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot:28]: Same as for previous test.

**Review by jiripetrlik — APPROVED** [review]

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot (hunk 6 above)
Comment (ChughShilpa) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-kueue-e2e-tests.robot:22]: I am thinking to add these tags later when kueue will be integrated with Distributed Workloads

**Review by sutaakar — APPROVED** [review]

**Pull request merged**